### ISSUE TITLE
Broken Windows build

### ISSUE THREAD

**Issue by cbarrynoble** [OpenZWave/node-openzwave-shared#187]
I'm using node-openzwave-shared via node-red-contrib-openzwave.

Yesterday, I updated node-red to 0.17.4 and node.js to 6.11.1. When I performed “npm rebuild” after upgrading node.js, the build failed with the following linker error:

OpenZWave.lib(Value.obj) : error LNK2001: unresolved external symbol "private: void __cdecl v8::Isolate::CollectAllGarbage(char const *)" (?CollectAllGarbage@Isolate@v8@@AEAAXPEBD@Z) 

Although I was using 1.1.6 of node-red-contrib-openzwave, an attempt to install the latest version failed the same way.

Is this a known problem? I understand that the maintainers may not be focusing on Windows so I’d like to know how I can help.

Thanks,
Barry

**Comment by HellandB** [comment]
I just set up a fresh environment for running node-red on a Windows 7 client (see installation steps at th end of the post). I am trying to install the node-red-contrib-openzwave, which is dependent on node-openzwave-shared. But encounter problems when tryint to install node-openzwave-shared.

```
C:\Users\xbmc\.node-red>npm i openzwave-shared

> openzwave-shared@1.4.0 preinstall C:\Users\xbmc\.node-red\node_modules\.stagin
g\openzwave-shared-fee7883d
> node lib/install-ozw.js

Installing Open Z-Wave
Installing dependencies to C:\Users\xbmc\AppData\Local\Temp\ozwinstall-8w0np8ogy
hm15drou5wmi
npm WARN deprecated wrench@1.5.9: wrench.js is deprecated! You should check out
fs-extra (https://github.com/jprichardson/node-fs-extra) for any operations you
were using wrench for. Thanks for all the usage over the years.
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\xbmc\AppData\L
ocal\Temp\ozwinstall-8w0np8ogyhm15drou5wmi\package.json'
npm WARN ozwinstall-8w0np8ogyhm15drou5wmi No description
npm WARN ozwinstall-8w0np8ogyhm15drou5wmi No repository field.
npm WARN ozwinstall-8w0np8ogyhm15drou5wmi No README data
npm WARN ozwinstall-8w0np8ogyhm15drou5wmi No license field.
Temp Path: C:\Users\xbmc\AppData\Local\Temp\ozwinstall-8w0np8ogyhm15drou5wmi
Install Path: C:\Users\xbmc\AppData\Local\OpenZWave
Downloading open zwave source.
Extracting open zwave source.
Building open zwave
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
C:\Users\xbmc\AppData\Local\Temp\ozwinstall-8w0np8ogyhm15drou5wmi\ozw\open-zwav
e-master\build\openzwave.vcxproj(20,3): error MSB4019: The imported project "C:
\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Impo
rt> declaration is correct, and that the file exists on disk.
C:\Users\xbmc\.node-red\node_modules\.staging\openzwave-shared-fee7883d\lib\inst
all-ozw.js:147
  throw err;
  ^

Error: `C:\Windows\Microsoft.NET\Framework\v4.0.30319\msbuild.exe` failed with e
xit code: 1
    at ChildProcess.onExit (C:\Users\xbmc\AppData\Local\Temp\ozwinstall-8w0np8og
yhm15drou5wmi\node_modules\node-gyp\lib\build.js:258:23)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
npm WARN node-red-project@0.0.1 No repository field.
npm WARN node-red-project@0.0.1 No license field.
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\
node_modules\\npm\\bin\\npm-cli.js" "i" "openzwave-shared"
npm ERR! node v6.11.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! openzwave-shared@1.4.0 preinstall: `node lib/install-ozw.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the openzwave-shared@1.4.0 preinstall script 'node lib/instal
l-ozw.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the openzwave-shared pack
age,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node lib/install-ozw.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs openzwave-shared
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls openzwave-shared
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\xbmc\.node-red\npm-debug.log

C:\Users\xbmc\.node-red>
```

My steps to reproduce the error
1. Download and install node.js
2. npm i -g node-red 
3. npm i -g --production windows-build-tools
4. npm i openzwave-shared

Let me know if you need the npm-debug.log or other information.

Cheers.

**Comment by jbreslin** [comment]
Did you manage to make any progress with these issues? 

I am struggling with a number of related issues and think that until unless it is brought up to date then support for Windows is as good as dropped and that would be a real pity.

If I can assist or offer to support anyone else with time or money then I will be happy to do so withing reason! ;)

My main concern is not to do anything that compromises the work of others.

**Comment by HellandB** [comment]
Nice to see interest for this topic. I am still hoping someone could provide some help to overcome these issues.

I am looking into upgrade my home with z-wave devices, but I need this node to work first..

**Comment by Apollon77** [comment]
I also use it in a project and activated the build on appveyor-ci (great WIndows CI tool!!) and my nuild get's killed by error:

..\src\utils.cc(331): error C2660: 'OZW::getValueIdDescriptor': function does not take 4 arguments [C:\projects\ioBroker.zwave\node_modules\openzwave-shared\build\openzwave_shared.vcxproj]

Full log:
https://ci.appveyor.com/project/Apollon77/iobroker-zwave/build/test-1/job/slv11x9i9h798r6h

New issue or connected to this? In the end windows build is broken ... :-(

**Comment by HellandB** [comment]
I don't know. I gave up since I haven't bought z-wave devices yet. Would be great if somone could come up with a solution though, on how to install this plugin on Windows. Then I can start build up on z-wave devices:)

**Comment by Apollon77** [comment]
ok, I created a new issue ... maybe some can tell what is connected :-) 
#211

**Comment by ekarak** [comment]
Version 1.4.3 just released. This includes the fix in #222 and adds Appveyor CI support for Windows builds.
